### ISSUE TITLE
Unknown type captured by wildcard pattern in match statement is not reported as error

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#4703]
When a `match` statement contains a wildcard capture and strict mode is enabled, the capture of an `Unknown` type by the wildcard does not currently result in any diagnostic. This can lead to type checking holes.

Consider the following:

```python
def to_bytes(value) -> bytes:
    match value:
        case _:
            return b""
```

**Comment by erictraut** [comment]
This is included in pyright 1.1.296, which I just published. It will also be included in a future release of pylance.
